app/Services/ScBookingService.cs

```diff
@@ -231,6 +231,12 @@ public async Task<List<int>> GetConferenceTypeIds(ScCaseSearchViewModel model)
             return await GetConferenceTypesAsync(model.CaseLocationName);
         }
 
+        // Returns booking types from the cache
+        public async Task<string[]> GetAvailableBookingTypes()
+        {
+            return await _cache.GetAvailableBookingTypesAsync();
```

Comment:
You could filter out unsupported types here so you don't have to do it in the controllers
```
var supportedTypes = ScHearingType.HearingTypeIdMap.Keys.Select(keyName => keyName);
return (await _cache.GetAvailableBookingTypesAsync())
    .ToList()
    .Intersect(supportedTypes)
    .ToArray();
```

Reply:
I just changed the cache function back so it stores the original array response from the API, and then I've got it filtering and converting it to a list here 👍